Mlem/API/Models/Posts/APIPost.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  APIPost.swift
+//  Mlem
+//
+//  Created by Nicholas Lawson on 09/06/2023.
+//
+
+import Foundation
+
+struct APIPost: Decodable {
+    let apId: String
+    let body: String?
+    let communityId: Int
+    let creatorId: Int
+    let deleted: Bool
+    let embedDescription: String?
+    let embedTitle: String?
+    let embedVideoUrl: String?
+    let featuredCommunity: Bool
+    let featuredLocal: Bool
+    let id: Int
+    let languageId: Int
+    let local: Bool
+    let locked: Bool
+    let name: String
+    let nsfw: Bool
+    let published: Date
+    let removed: Bool
+    let thumbnailUrl: URL?
+    let updated: String?
```

Comment:
Is this not a date?

Reply:
Just checked, this is an optional date:

```
  pub published: chrono::NaiveDateTime,
  pub updated: Option<chrono::NaiveDateTime>,
```

Reply:
it is, yeah - the front-end doesn't use this property yet so it's just modelled based on the type returned, I'll change those fields to be correct here and on the other models

Reply:
addressed in 77b1f8cd998bd6c3d4209c7380cff943f4ea7932